internal/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,73 @@
+package dispatcher
+
+import (
+	"context"
+
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+)
+
+type Dispatcher interface {
+	CheckDispatcher
+}
+
+// CheckDispatcher defines an interface that can be implemented to resolve dispatched Check
+// requests. Implementations of the interface can provide local or remote dispatching mechansims.
+type CheckDispatcher interface {
```

Comment:
Eventually we'll implement this interface over an RPC medium and each OpenFGA server will have service discovery to discover other OpenFGA replicas so that each one can dispatch/distribute requests to one another. It doesn't really make sense to introduce this until we have caching to go along with it, because otherwise we'll just be adding a lot of network traffic and extra latency for little to no benefit.